backend/infrahub/core/migrations/query/attribute_add.py

```diff
@@ -0,0 +1,82 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Optional
+
+from infrahub.core.constants import NULL_VALUE, RelationshipStatus
+from infrahub.core.query import Query
+
+if TYPE_CHECKING:
+    from infrahub.database import InfrahubDatabase
+
+
+class AttributeAddQuery(Query):
+    name = "attribute_add"
+
+    def __init__(
+        self,
+        node_kind: str,
+        attribute_name: str,
+        attribute_kind: str,
+        branch_support: str,
+        default_value: Optional[Any] = None,
+        **kwargs: Any,
+    ) -> None:
+        self.node_kind = node_kind
+        self.attribute_name = attribute_name
+        self.attribute_kind = attribute_kind
+        self.branch_support = branch_support
+        self.default_value = default_value
+
+        super().__init__(**kwargs)
+
+    async def query_init(self, db: InfrahubDatabase, **kwargs: dict[str, Any]) -> None:
+        branch_filter, branch_params = self.branch.get_query_filter_path(at=self.at.to_string())
+        self.params.update(branch_params)
+
+        self.params["node_kind"] = self.node_kind
+        self.params["attr_name"] = self.attribute_name
+        self.params["attr_type"] = self.attribute_kind
+        self.params["branch_support"] = self.branch_support
+
+        if self.default_value:
```

Comment:
`if self.default_value is not None`?

Reply:
I might have been a bit unclear in the first comment but I think this should be changed. If the default value is `0` we will set it to `NULL_VALUE` instead with this.